crates/backend/src/codegen.rs

```diff
@@ -971,22 +971,42 @@ impl TryToTokens for ast::ImportFunction {
                 );
             }
             Some(ref ty) => {
-                abi_ret = quote! {
-                    <#ty as wasm_bindgen::convert::FromWasmAbi>::Abi
-                };
-                convert_ret = quote! {
-                    <#ty as wasm_bindgen::convert::FromWasmAbi>
-                        ::from_abi(#ret_ident)
-                };
+                if self.function.r#async {
+                    abi_ret = quote! { js_sys::Promise };
```

Comment:
I think this will want to be `<js_sys::Promise as FromWasmAbi>::Abi` perhaps?

Reply:
Right.